Saule/Queries/Pagination/IPagedResult.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections;
+
+namespace Saule.Queries.Pagination
+{
+    /// <summary>
+    /// Paged result that can be used to tell Saule how many pages we have
+    /// </summary>
+    public interface IPagedResult
```

Comment:
I don't really understand why this needs to be an interface. I'd rather it's just a class so we can extend it later without breaking implementations.